examples/api/test/widgets/text_magnifier/text_magnifier.0_test.dart

```diff
@@ -73,17 +73,35 @@ void main() {
 
     final TestGesture gesture = await tester.startGesture(handlePos);
 
-    await gesture.moveTo(_textOffsetToPosition(
-      tester,
-      example.MyApp.textFieldText.length - 2,
-    ));
+    await gesture.moveTo(
+      _textOffsetToPosition(
+        tester,
+        example.MyApp.textFieldText.length - 2,
+      ),
+    );
     await tester.pump();
+  }
+
+  testWidgets('should show custom magnifier on drag', (WidgetTester tester) async {
+    await tester.pumpWidget(const example.MyApp());
 
+    await showMagnifier(tester);
     expect(find.byType(example.CustomMagnifier), findsOneWidget);
 
     await expectLater(
       find.byType(example.MyApp),
       matchesGoldenFile('text_magnifier.0_test.png'),
     );
-  }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS,  TargetPlatform.android }));
+  }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS, TargetPlatform.android }));
+
+
+  for (final TextDirection textDirection in TextDirection.values) {
+    testWidgets('should show custom magnifier in $textDirection', (WidgetTester tester) async {
```

Comment:
The text (`'Custom Magnifier!'`) mostly consists of characters of strong LTR directionality (except for the space in between and the exclamation mark) so most of them will still be rendered from left to right.

Reply:
I added a strong RTL string so we can test RTL.